opentracing/span.go

```diff
@@ -0,0 +1,91 @@
+package opentracing
+
+import "golang.org/x/net/context"
+
+type Span interface {
+	// Creates a child span, optionally modifying a `context.Context` parent.
+	//
+	// `parent` is optional; if specified, it is used as the descendant for the
+	// returned `context.Context` object.
+	//
+	// Regardless of whether `parent` is specified, `StartChildSpan` returns a
+	// `Span` that descends directly from the callee. The returned
+	// `context.Context` instance derives from `parent` if.f. it was specified.
+	StartChildSpan(operationName string, parent ...context.Context) (Span, context.Context)
+
+	// Adds a tag to the span. The `value` is immediately coerced into a string
+	// using fmt.Sprint().
+	//
+	// If there is a pre-existing tag set for `key`, it is overwritten.
+	SetTag(key string, value interface{})
+
+	// `Message` is a format string and can refer to fields in the payload by path, like so:
+	//
+	//   "first transaction is worth ${transactions[0].amount} ${transactions[0].currency}"
+	//
+	// , and the payload might look something like
+	//
+	//   map[string]interface{}{
+	//       transactions: map[string]interface{}[
+	//           {amount: 10, currency: "USD"},
+	//           {amount: 11, currency: "USD"},
+	//       ]}
+	Info(message string, payload ...interface{})
+
+	// Like Info(), but for warnings.
+	Warning(message string, payload ...interface{})
+
+	// Like Info(), but for errors.
+	Error(message string, payload ...interface{})
+
+	// Sets the end timestamp and calls the `Recorder`s RecordSpan() internally.
+	//
+	// Calling Finish() multiple times on the same Span instance may lead to
+	// undefined behavior.
```

Comment:
I would make an even stronger statement - Finish() should be the last call on the span, calling any of the above methods after Finish() has been called has undefined behavior and may be ignored by the tracing system.